src/components/AnonymousChat/child/ChatItemTargetText.tsx

```diff
@@ -1,100 +1,121 @@
+
+  const animatedBubbleStyle = useAnimatedStyle(() => ({
+    transform: [{translateX: bubblePosition.value}]
+  }));
+
+  const contextMenuActions = [
+    {title: 'Reply', systemIcon: 'arrow.turn.up.left'},
+    {title: 'Copy Message', systemIcon: 'square.on.square'}
+  ];
+
   const renderAvatar = React.useCallback(() => {
-    if (isContinuous) return <View style={styles.avatar} />;
+    if (!isShowUserInfo) return <View style={styles.avatar} />;
     return <View style={styles.mr8}>{avatar}</View>;
   }, []);
 
+  const onSwipeToReply = (direction) => {
+    if (direction === 'right') return;
+    if (swipeableRef.current) swipeableRef.current?.close();
+    setReplyPreview({
+      username,
+      time,
+      message,
+      messageId: data?.id,
+      chatType,
+      messageType: 'regular'
+    });
+  };
+
   return (
-    <View style={styles.chatContainer}>
-      {renderAvatar()}
-      <View style={styles.textContainer}>
-        {!isContinuous && (
-          <View testID="chat-item-user-info" style={styles.chatTitleContainer}>
-            <Text style={styles.userText}>{username}</Text>
-            <View style={styles.dot} />
-            <Text style={styles.timeText}>{time}</Text>
-          </View>
-        )}
-        <Text testID="chat-item-message" style={styles.text}>
-          {message}
-        </Text>
+    <Swipeable
+      testID="swipeable"
+      ref={swipeableRef}
+      friction={3}
+      overshootFriction={2}
+      onSwipeableOpen={onSwipeToReply}
+      renderLeftActions={replyIcon}>
+      <View style={containerStyle(false, isReplyPrompt)}>
+        <Animated.View style={[styles.wrapper, animatedBubbleStyle]}>
+          {renderAvatar()}
+
+          <ContextMenu
+            previewBackgroundColor="transparent"
+            style={{flex: 1}}
+            actions={contextMenuActions}
+            onPress={(e) => onContextMenuPressed(e, data, chatType)}>
+            <View style={styles.radius8}>
+              <View style={textContainerStyle(false)}>
+                {isShowUserInfo && (
+                  <View testID="chat-item-user-info" style={styles.chatTitleContainer}>
+                    <Text style={[styles.userText, textStyle(false)]}>{username}</Text>
+                    <View style={dotStyle(false)} />
+                    <Text style={[styles.timeText, textStyle(false)]}>{time}</Text>
+                  </View>
+                )}
+                <ChatReplyView
+                  type={chatType}
+                  messageType={messageType}
+                  replyData={data?.reply_data}
+                />
+                <Text testID="chat-item-message" style={messageStyle(false, messageType)}>
+                  {message}
+                </Text>
+              </View>
+            </View>
+          </ContextMenu>
+        </Animated.View>
       </View>
-    </View>
+    </Swipeable>
   );
 };
 
-export default ChatItemTargetText;
+export default React.memo(ChatItemTargetText);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes in this file are quite extensive and involve a lot of refactoring. The new code introduces swipeable behavior for replying to messages, context menu actions, and animations. It also uses hooks and memoization to optimize performance. However, there are a few areas that could be improved:

1. Error handling: There is no error handling in the `onSwipeToReply` function. If an error occurs while setting the reply preview or closing the swipeable, it will cause the app to crash. Consider adding try-catch blocks around these operations.

2. Code readability: The `ChatItemTargetText` component has become quite large and complex with many responsibilities. Consider breaking it down into smaller, more manageable components to improve readability and maintainability.

3. Type safety: The `direction` parameter in the `onSwipeToReply` function does not have a type. Consider defining a type for it to ensure type safety.

Here are some suggestions on how to improve these areas:

```diff
  // Add error handling to onSwipeToReply
  const onSwipeToReply = (direction: 'left' | 'right') => {
    if (direction === 'right') return;
    try {
      if (swipeableRef.current) swipeableRef.current?.close();
      setReplyPreview({
        username,
        time,
        message,
        messageId: data?.id,
        chatType,
        messageType: 'regular'
      });
    } catch (error) {
      console.error('Failed to reply:', error);
    }
  };

  // Break down ChatItemTargetText into smaller components
  const UserInfo = ({username, time}) => (
    <View testID="chat-item-user-info" style={styles.chatTitleContainer}>
      <Text style={[styles.userText, textStyle(false)]}>{username}</Text>
      <View style={dotStyle(false)} />
      <Text style={[styles.timeText, textStyle(false)]}>{time}</Text>
    </View>
  );

  const Message = ({messageType, message}) => (
    <Text testID="chat-item-message" style={messageStyle(false, messageType)}>
      {message}
    </Text>
  );

  // Then use these components in ChatItemTargetText
  return (
    <Swipeable
      /* ... */
    >
      <View style={containerStyle(false, isReplyPrompt)}>
        <Animated.View style={[styles.wrapper, animatedBubbleStyle]}>
          {renderAvatar()}
          <ContextMenu
            /* ... */
          >
            <View style={styles.radius8}>
              <View style={textContainerStyle(false)}>
                {isShowUserInfo && <UserInfo username={username} time={time} />}
                <ChatReplyView
                  type={chatType}
                  messageType={messageType}
                  replyData={data?.reply_data}
                />
                <Message messageType={messageType} message={message} />
              </View>
            </View>
          </ContextMenu>
        </Animated.View>
      </View>
    </Swipeable>
  );
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->